Cython/Compiler/PyrexTypes.py

```diff
@@ -4707,6 +4707,13 @@ class PythonTypeConstructorMixin:
     """Used to help Cython interpret indexed types from the typing module (or similar)
     """
     modifier_name = None
+    contains_none = False
+
+    def is_optional(self):
```

Comment:
`is_optional()` feels unclear to me as a name. It doesn't strictly refer to `typing.Optional`, just to something that is "optional" in whatever way.

Isn't a name like `allows_none` more concrete for the purpose?

Reply:
> Isn't a name like allows_none more concrete for the purpose?

I am not good in naming things so I am totally fine on whatever has consensus among Core devs.

Reply:
fixed.